### PR TITLE
adding new rating field

### PR DESCRIPTION
**Pull request by iruzevic**
# Description

- added new rating field
- moving custom attributes and custom props from PHP to manifest for easier sharing
- defining internal field types for easier usage

# Screenshots / Videos

Screenshot 2023-11-20 at 10 55 13
Screenshot 2023-11-20 at 10 55 15
Screenshot 2023-11-20 at 10 55 18

### CODE REVIEW

**Review by goranalkovic-infinum — APPROVED** [review]

File: src/Blocks/components/rating/rating.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+$ratingAttrs = Components::checkAttr('ratingAttrs', $attributes, $manifest);
+$ratingFieldAttrs = Components::checkAttr('ratingFieldAttrs', $attributes, $manifest);
+$ratingAmount = Components::checkAttr('ratingAmount', $attributes, $manifest);
+$ratingHideLabel = false;
+
+$ratingClass = Components::classnames([
+	Components::selector($componentClass, $componentClass),
+	Components::selector($componentJsClass, $componentJsClass),
+	Components::selector($additionalClass, $additionalClass),
+]);
```
Comment (kancijan) [inline src/Blocks/components/rating/rating.php:42]:
```suggestion
$ratingClass = Components::classnames([
	$componentClass,
	$componentJsClass,
	$additionalClass,
]);
```

File: src/Blocks/components/rating/rating.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+$ratingStarClass = Components::classnames([
+	Components::selector($componentClass, $componentClass, 'star'),
+]);
```
Comment (kancijan) [inline src/Blocks/components/rating/rating.php:46]:
```suggestion
$ratingStarClass = Components::selector($componentClass, $componentClass, 'star');
```

**Review by iobrado — APPROVED** [review]

**Review by kancijan — CHANGES_REQUESTED** [review]

**Review by dadadavorin — APPROVED** [review]

File: src/Blocks/components/form/assets/enrichment.js [hunk 3]
```diff
@@ -294,22 +294,22 @@ export class Enrichment {
...
 					this.utils.setManualSelectValue(formId, name, value);
 					break;
-				case StateEnum.TYPE_INT_CHECKBOX:
+				case this.state.getStateIntType('checkbox'):
 					this.utils.setManualCheckboxValue(formId, name, value);
 					break;
-				case StateEnum.TYPE_INT_RADIO:
-					this.utils.setManualRadioValue(formId, name, value);
+				case this.state.getStateIntType('rating'):
+					this.utils.setManualRatingValue(formId, name, value);
```
Comment (piqusy) [inline src/Blocks/components/form/assets/enrichment.js:312]: Radio not needed anymore?

File: src/Blocks/components/rating/components/rating-editor.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		<>
+			<FieldEditor
+				{...props('field', attributes, {
+					fieldContent: rating,
+					fieldIsRequired: checkAttr('ratingIsRequired', attributes, manifest),
+				})}
+				additionalFieldClass={additionalFieldClass}
+				selectorClass={componentName}
+			/>
+		</>
```
Comment (piqusy) [inline src/Blocks/components/rating/components/rating-editor.js:56]:
```suggestion
			<FieldEditor
				{...props('field', attributes, {
					fieldContent: rating,
					fieldIsRequired: checkAttr('ratingIsRequired', attributes, manifest),
				})}
				additionalFieldClass={additionalFieldClass}
				selectorClass={componentName}
			/>
```

**Review by piqusy — CHANGES_REQUESTED** [review]
Nice work 👏🏼 

I left a suggestion and one question

File: src/Blocks/components/form/assets/enrichment.js (hunk 3 above)
Comment (iruzevic) [reply src/Blocks/components/form/assets/enrichment.js:312]: dam boy gooooooood catch, tnx

File: src/Blocks/components/rating/components/rating-editor.js (hunk 4 above)
Comment (iruzevic) [reply src/Blocks/components/rating/components/rating-editor.js:56]: tnx

File: src/Blocks/components/rating/rating.php (hunk 1 above)
Comment (iruzevic) [reply src/Blocks/components/rating/rating.php:42]: Tnx for the suggestion but I will leave it as it

File: src/Blocks/components/rating/rating.php (hunk 2 above)
Comment (iruzevic) [reply src/Blocks/components/rating/rating.php:46]: tnx, changed

**Review by piqusy — APPROVED** [review]

**Review by kancijan — APPROVED** [review]

**Pull request merged**